src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.FeatureManagement.FeatureFilters;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights
+{
+    /// <summary>
+    /// Provides extension methods for tracking events with TargetingContext.
+    /// </summary>
+    public static class ApplicationInsightsTelemetryExtensions
+    {
+        /// <summary>
+        /// Extension method to track an event with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, string eventName, TargetingContext targetingContext, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null)
+        {
+            if (properties == null)
+            {
+                properties = new Dictionary<string, string>();
+            }
+
+            properties["TargetingId"] = targetingContext.UserId;
+
+            telemetryClient.TrackEvent(eventName, properties, metrics);
+        }
+
+        /// <summary>
+        /// Extension method to track an <see cref="EventTelemetry"/> with <see cref="TargetingContext"/>.
+        /// </summary>
+        public static void TrackEvent(this TelemetryClient telemetryClient, EventTelemetry telemetry, TargetingContext targetingContext)
+        {
+            if (telemetry == null)
+            {
+                telemetry = new EventTelemetry();
+            }
+
+            telemetry.Properties["TargetingId"] = targetingContext.UserId;
```

Comment:
if `targetingContext` is null, this is a null-ref. I recommend we require it for this overload.